File: fetch_issues.py
import requests
from jinja2 import Template

GITHUB_REPO = 'just-koala/just-koala.github.io'  # Replace 'username' with your GitHub username
GITHUB_TOKEN = 'KOALA_TOKEN'

def fetch_issues(repo):
    url = f'https://api.github.com/repos/{repo}/issues'
    headers = {
        'Authorization': f'token {GITHUB_TOKEN}'
    }
    response = requests.get(url, headers=headers)
    response.raise_for_status()
    return response.json()

def generate_post(issue):
    template = Template(open('template.html').read())
    return template.render(title=issue['title'], content=issue['body'])

def main():
    issues = fetch_issues(GITHUB_REPO)
    for issue in issues:
        with open(f'_posts/{issue["title"].replace(" ", "_")}.html', 'w') as f:
            f.write(generate_post(issue))

if __name__ == '__main__':
    main()

File: fetch_issues.cs
using System;
using System.IO;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;
using HtmlAgilityPack;

// Configuration
const string GITHUB_REPO = "your-username/your-repo-name";
string GITHUB_TOKEN = Environment.GetEnvironmentVariable("KOALA_TOKEN");

// Fetch issues from GitHub
async Task<JArray> FetchIssues(string repo)
{
    string url = $"https://api.github.com/repos/{repo}/issues";
    using var client = new HttpClient();
    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("token", GITHUB_TOKEN);
    client.DefaultRequestHeaders.UserAgent.ParseAdd("Mozilla/5.0 (compatible; GrandCircus/1.0)");
    var response = await client.GetStringAsync(url);
    return JArray.Parse(response);
}

// Generate blog post from issue
string GeneratePost(JObject issue)
{
    string template = File.ReadAllText("template.html");
    string title = issue["title"].ToString();
    string body = issue["body"].ToString();

    var doc = new HtmlDocument();
    doc.LoadHtml(template);
    doc.DocumentNode.SelectSingleNode("//title").InnerHtml = title;
    doc.DocumentNode.SelectSingleNode("//h1").InnerHtml = title;
    doc.DocumentNode.SelectSingleNode("//div").InnerHtml = body;

    return doc.DocumentNode.OuterHtml;
}

// Main function
async Task Main()
{
    var issues = await FetchIssues(GITHUB_REPO);
    Directory.CreateDirectory("posts");

    foreach (JObject issue in issues)
    {
        if (issue["labels"].ToString().Contains("blog"))
        {
            string filename = $"posts/{issue["title"].ToString().Replace(' ', '_')}.html";
            File.WriteAllText(filename, GeneratePost(issue));
        }
    }
}

await Main();
